### DISCUSSION TITLE
Can WebLate commit changes of all components as one united commit?

### DISCUSSION THREAD

**Discussion by Wohlstand** [WeblateOrg/weblate#11138] [Q&A] [answered]
Hello!

Recently, I switched from the "pull-request" style into "direct commit", and my friend mentioned that there are 4 commits sent at once (because of separated components being updated). Is possible to let WebLate to commit changes of all components as one single commit rather than separated commits per every component?

**Comment by nijel** [comment, marked as answer]
Weblate always commits changes individually to properly credit translators. If your project has different requirements, you can use https://docs.weblate.org/en/latest/admin/addons.html#squash-git-commits.

**Reply by Wohlstand** [reply]
If the work on all components done by same person, then that should be squashed, yeah. But if different persons done their work, then that's should be reasonable.

**Reply by Wohlstand** [reply]
I.e. I am fine to have separated commits as long as they done by different translators. But when the work on different components done by the same person, then that would just cause a trashing of the repo by an extra pile of commits. Anyway thanks, I'll try to checkout the squash add-on.

**Reply by Wohlstand** [reply]
Okay, I checked out the thing, seems looks that what I needed, thank you a lot!
